amaranth/lib/data.py

```diff
@@ -789,6 +799,221 @@ def __repr__(self):
         return f"{self.__class__.__name__}({self.__layout!r}, {self.__target!r})"
 
 
+class Const(ValueCastable):
+    """A constant value viewed through the lens of a layout.
+
+    The :class:`Const` class is a constant analog of :class:`View`. It implements largely
+    the same operations, except that it deals with compile-time constants instead of arbitrary
+    :class:`Value`.
+
+    Creating a const
+    ################
+
+    A constant can be created from a dict or list of field values using :meth:`Layout.const`,
+    or from a bit-pattern using :meth:`Layout.from_bits`.
+
+    Accessing a const
+    #################
+
+    Slicing a const or accessing its attributes returns a part of the underlying value
+    corresponding
+    to the field with that index or name. If the shape of the field is a :class:`Layout`,
+    the accessed value will be another :class:`Const`. If it is another :ref:`shape-like <lang-shapelike>`
+    object implementing ``from_bits``, it will be the result of calling that method. Otherwise,
+    it will be a plain :class:`int`.
+
+    Slicing a view whose layout is an :class:`ArrayLayout` can be done with an index that is
+    an Amaranth value instead of a constant integer. The returned element is chosen dynamically
+    in that case, and the resulting value will be a :class:`View` instead of :class:`Const`.
+
+    A const can be compared for equality with another const or view of the same layout.
+    When compared with another const, the result will be a Python :class:`bool`.
+    When compared with a view, the result will be a single-bit value. No other operators
```

Comment:
Here the disambiguation is actually important!

```suggestion
    When compared with a view, the result will be a single-bit :class:`Value`. No other operators
```